ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py

```diff
@@ -759,3 +759,75 @@ def test_torch_solve(
         input=input,
         other=other,
     )
+
+
+# tensorinv
+@st.composite
+def _tensorinv_helper(draw):
+    def factors(x):
+        result = [1, ]
+        i = 2
+        while i * i <= x:
+            if x % i == 0:
+                result.append(i)
+                if x // i != i:
+                    result.append(x // i)
+            i += 1
+        result.append(x)
+        return np.array(result)
+    ind = draw(helpers.ints(min_value=1, max_value=6))
+    product_half = draw(helpers.ints(min_value=2, max_value=25))
+    factors_list = factors(product_half)
+    shape = ()
+    while len(shape) < ind and ind > 2:
+        while np.prod(shape) < product_half:
+            a = factors_list[np.random.randint(len(factors_list))]
+            shape += (a,)
+        if np.prod(shape) > product_half or len(shape) > ind:
+            shape = ()
+        while len(shape) < ind and shape != ():
+            shape += (1,)
+        if np.prod(shape) == product_half:
+            shape += shape[::-1]
+            break
+    if ind == 1 and shape == ():
+        shape += (product_half, product_half)
+    if ind == 2 and shape == ():
+        shape += (1, product_half, product_half, 1)
+    shape_cor = ()
+    for i in shape:
+        shape_cor += (int(i),)
+    dtype = draw(helpers.get_dtypes("float"))
+    input = np.random.rand(*shape_cor)
```

Comment:
About the second case, I think there are a couple of possible ways around it. 
1. You could define a simple helper function like
```
def _check_invertibility(x, ind):
    try:
        np.linalg.tensorinv(x, ind=ind)
    except:
        return False
    return True
```
and then use the `array_values` helper with `.filter(lambda x: _check_invertibility(x[1], ind)`.
2. Does `np.linalg.tensorinv` ever cause an exception for 1-d inputs (or, equivalently, would we ever consider a 1-d array as non-invertible?). If not, then you could use the `array_values` helper with something like `.filter(lambda x: np.linalg.cond(x[1]) < 1 / sys.float_info.epsilon if len(x[1].shape) > 1 else True`.

Reply:
Fyi, the reason I insist on trying to find a way to use our hypothesis-based helpers is that hypothesis keeps an example database that helps a lot with debugging. 
Quoting from the documentation: "When Hypothesis finds a bug it stores enough information in its database to reproduce it. This enables you to have a classic testing workflow of find a bug, fix a bug, and be confident that this is actually doing the right thing because Hypothesis will start by retrying the examples that broke things last time.". 
Even if a function passes all tests when merged to master, it's possible it might fail in the future due to a change in another part of our code. In this case it would be best if we could debug it efficiently using the hypothesis test failure logs.

Reply:
I wonder why are we using `np.linalg.tensorinv` inside the test function that testing if four versions of this function gives same result. Isn't it against the purpose of the test. Still, I tried them.

1. Also, I didn't understand why we are using `x[1]` but here's the result. 

![pycharm64_5DYdU9R5Id](https://user-images.githubusercontent.com/51514616/211522523-07d408b6-da42-4a0b-882b-15da460f2664.png)
result:
![pycharm64_2CPie18owo](https://user-images.githubusercontent.com/51514616/211522585-1dcc8ad2-bc03-4f46-a1ac-b5c6630bd81c.png)
I changed `x[1]` to `x`:
![pycharm64_N1TumRn14R](https://user-images.githubusercontent.com/51514616/211522716-0bce9011-5c42-4faf-87db-b5a8ffb0b84f.png)
2.
When we try to send any shape doesn't satisfy product condition, it throws error. 
Ex:
```py
np.linalg.tensorinv(np.random.rand(1))
```
This is valid because `np.prod(shape[:ind] == np.prod(shape[ind:])` for any value of positive integer `ind`. But when we try:
```py
np.linalg.tensorinv(np.random.rand(2))
```
This throws an error that saying last 2-d is not square. My shape generator never creates a 1-d array. Cause only valid case is 1d array with 1 element and no coupling between `ind` values for this case. But since the first one didn't work, I didn't try this solution.

Reply:
Alright, I will experiment a bit with it locally and if nothing works for me either, we will merge it as it is.

Reply:
![image](https://user-images.githubusercontent.com/111577222/211547085-0e424c76-2d53-4968-b21f-3594c6203efa.png)
![image](https://user-images.githubusercontent.com/111577222/211547220-0d539a0e-40ed-4753-b87e-8f0479f1cff4.png)
![image](https://user-images.githubusercontent.com/111577222/211547291-e213da93-0800-470d-9b5f-7d0c24d97e84.png)
With the above changes the test passes locally. It's a bit hacky to use the `try-except` in `test_torch_tensorinv` but I think it's still preferable as it allows us to use the hypothesis helper.

Reply:
From my understanding, you bypass failed cases. In my humble opinion, this can cause passing the test for a set of values which shouldn't be.  Adding a keyword argument to the helper for this function might solve the issue? I can try to work on it after adding a `TODO` to this `_tensorinv_helper`. In case you want to proceed with your solution, I can push commits.

Reply:
Regardless of how we test it, you assert inversiblity inside `tensorinv`, which means that we expect `tensorinv` to work only for invertible inputs (if I understand correctly). So, I think there is no harm in skipping the cases where the inversibility assertion gets triggered. Is there another case where this would happen " this can cause passing the test for a set of values which shouldn't be"?

Reply:
But we are not checking invertibility of the input as it is. We reshape the input and then checking invertibility (which original function also does the same). I guess that's why we can't filter them in the first place. The equation to hold is the dot product of inverted tensor with itself should be equal to identity of the space. Let me try to explain with an example:
```py
# ind = 2, shape = (8, 3, 4, 6), a is input, b is the one we check if inverse exist
a = np.random.rand(8, 3, 4, 6)
prod_ind_end = np.prod(a.shape[:2])
print('prod_ind_end', prod_ind_end)
b = a.reshape(prod_ind_end, -1)
print('shape a', a.shape, '\n', 'shape b', b.shape)
# np.linalg.inv(a)
np.linalg.inv(b)
```
If you run this first, you'll see no error. Then remove `np.linalg.inv(b)` and  `#`, it will throw an error.  Any of determinant calculation also won't work because of the same reason. We need to check the determinant of the reshaped tensor. 

I just tried a filter by drawing a square shape and reshaping after that due to my generated shape, but it didn't work again. The main difficulty to test invertibility is that. Almost all the (probably all) input tensors are not invertible (correct way to say this, we can't check their invertibility with the functions we have. `det`, `inv` etc.) for dimensions higher than 2. We need a function that calculates determinant or inverse, w.r.t given axis. I also checked this but couldn't find.

Reply:
But if we do 
```
    except Exception as e:
        if hasattr(e, "message"):
            if 'Reshaped tensor is not invertible' in e.message:
                assume(False)
```
then we only skip the cases where `tensorinv` is not supposed to work. We assume False only for the exception thrown by your `tensorinv` code.

Reply:
> Is there another case where this would happen " this can cause passing the test for a set of values which shouldn't be"?

Yes, there might be. Explanation is in the previous comment. I didn't mean error message is wrong. We completely ignore this error, which should be there for a reason. 

For now, I just pushed commits with your correction. I'll think about this, and if I found a solution, I can create a PR and inform you.